### PR TITLE
PMM-3449 don't exit on unexpected payload

### PR DESCRIPTION
**Pull request by jprukner**

### CODE REVIEW

**Review by BupycHuk — APPROVED** [review]

File: client/channel/channel.go [hunk 1]
```diff
@@ -289,23 +309,45 @@ func (c *Channel) subscribe(id uint32) chan agentpb.ServerResponsePayload {
...
+}
+
+// cancel sends an error to the subscriber.
+func (c *Channel) cancel(id uint32, err error) {
+	if ch, _ := c.removeResponseChannel(id); ch != nil {
+		ch <- Response{Error: err}
+	}
+}
+
+func (c *Channel) publish(id uint32, status *protostatus.Status, resp agentpb.ServerResponsePayload) {
```
Comment (palash25) [inline client/channel/channel.go:337]: nit: lets put publish and subscribe methods next to each other so that the code is easier to understand

Comment (jprukner) [reply client/channel/channel.go:332]: Done

**Review by palash25 — APPROVED** [review]

File: client/channel/channel.go [hunk 2]
```diff
@@ -253,23 +268,48 @@ func (c *Channel) runReceiver() {
...
+		return nil
+	}
+	delete(c.responses, id)
+	return ch
+}
+
+// cancel sends an error to the subscriber.
+func (c *Channel) cancel(id uint32, err error) {
+	if ch := c.removeResponseChannel(id); ch != nil {
+		ch <- Response{Error: err}
```
Comment (BupycHuk) [inline client/channel/channel.go:307]: btw, shouldn't we close ch here?

**Review by BupycHuk — APPROVED** [review]

File: client/channel/channel.go (hunk 2 above)
Comment (jprukner) [reply client/channel/channel.go:314]: done

File: client/channel/channel.go [hunk 3]
```diff
@@ -140,16 +149,21 @@ func (c *Channel) Requests() <-chan *ServerRequest {
...
+		msg.Payload = resp.Payload.AgentMessageResponsePayload()
+	}
+	if resp.Status != nil {
+		msg.Status = resp.Status.Proto()
 	}
 	c.send(msg)
 }
 
 // SendAndWaitResponse sends request to pmm-managed, blocks until response is available, and returns it.
 // Response will be nil if channel is closed.
```
[the remark this answers is not in the document]
Comment (jprukner) [reply client/channel/channel.go]: done

**Pull request merged**